urwid/widget/pile.py

```diff
@@ -693,6 +703,55 @@ def _get_fixed_rows_sizes(
             tuple(w_h_args[idx] for idx in range(len(w_h_args))),
         )
 
+    def _get_flow_rows_sizes(
+        self,
+        size: tuple[int],
+        focus: bool = False,
+    ) -> tuple[Sequence[int], Sequence[int], Sequence[tuple[int] | tuple[()]]]:
+        """Get rows widths, heights and render size parameters
+
+        Flow case is the simplest one: minimum cycles in the logic and no widgets manipulation.
+        Here we can make some shortcuts
+        """
+        maxcol = size[0]
+        if not self.contents:
+            return (maxcol,), (0,), ()
+
+        widths: Sequence[int] = (maxcol,) * len(self.contents)
+        heights: list[int] = []
+        w_h_args: list[tuple[int, int] | tuple[int] | tuple[()]] = []
+        focus_position = self.focus_position
+
+        for i, (w, (f, height)) in enumerate(self.contents):
+            if isinstance(w, Widget):
+                w_sizing = w.sizing()
+            else:
```

Comment:
I hope to kill this warning in the next major release.
Following documentation for 2.0, all widgets have to subclass `Widget`, but some users not adopted code due to no warnings from the urwid side